### PR TITLE
Replace dockerfile for backend with multistage build

### LINKED ISSUE #27: Dockerfiles for Backend
- [x] Dont use alpine linux as base
- [ ] Use multistage builds to spereate dependencies from production build

### PR DESCRIPTION
**Pull request by Paulomart**
Closes #27

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
-# Remove dev-dependencies
-RUN npm ci --only=production && npm cache clean --force 
+FROM base AS release
+
+WORKDIR /app
+
+# If additional files are required, they can be copied from the builder layer to the release layer.
+COPY --from=builder /app/package*.json ./
+COPY --from=builder /app/dist ./dist
+COPY --from=builder /app/node_modules ./node_modules
```
Comment (timvahlbrock) [inline backend/Dockerfile:25]: Doesn't this also copy all dev dependencies?

Comment (Paulomart) [reply backend/Dockerfile:25]: Yes

Comment (timvahlbrock) [reply backend/Dockerfile:25]: Doesn't this bloat the image (a bit)?

Comment (Paulomart) [reply backend/Dockerfile:25]: Yeah its not optimal for image size but its faster for build time. Depends on what we are trying to optimize here..

**Pull request merged**